bundles/org.openhab.binding.yioremote/src/main/java/org/openhab/binding/yioremote/internal/YIOremoteDockHandler.java

```diff
@@ -0,0 +1,388 @@
+                } else if (yioremotedockactualstatus.equals(YIOREMOTEDOCKHANDLESTATUS.AUTHENTICATION_PROCESS)) {
+                    if (yioremotedockhandler.getbooleanauthenticationok()) {
+                        yioremotedockhandler.yioremotedockactualstatus = YIOREMOTEDOCKHANDLESTATUS.AUTHENTICATION_COMPLETE;
+                        updateStatus(ThingStatus.ONLINE);
+                        yioremotedockhandler.startwebsocketpollingthread();
+                        if (authenticationjob != null) {
+                            authenticationjob.cancel(true);
+                        }
+                    } else {
+                        yioremotedockhandler.yioremotedockactualstatus = YIOREMOTEDOCKHANDLESTATUS.AUTHENTICATION_FAILED;
+                    }
+                } else {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                            "Connection lost no ping from YIO DOCK");
+                    updateState(GROUP_OUTPUT, YIODOCKSTATUS, UnDefType.UNDEF);
+                }
+            }
+        };
+        authenticationjob = scheduler.scheduleWithFixedDelay(authenticationhread, 0, 5, TimeUnit.SECONDS);
+    }
+
+    private void startwebsocketpollingthread() {
+        Runnable websocketpollingthread = new Runnable() {
+            @Override
+            public void run() {
+                if (yioremotedockactualstatus.equals(YIOREMOTEDOCKHANDLESTATUS.AUTHENTICATION_COMPLETE)) {
+                    if (yioremotedockhandler.getbooleanheartbeat()) {
+                        logger.debug("heartbeat ok");
+                        yioremotedockhandler.sendMessage(YIOREMOTEMESSAGETYPE.HEARTBEAT, "");
+                    } else {
+                        yioremotedockactualstatus = YIOREMOTEDOCKHANDLESTATUS.CONNECTION_FAILED;
+                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                                "Connection lost no ping from YIO DOCK");
+                        updateState(GROUP_OUTPUT, YIODOCKSTATUS, UnDefType.UNDEF);
+                        if (websocketpollingjob != null) {
+                            websocketpollingjob.cancel(true);
+                        }
+                    }
+                } else {
+                    if (websocketpollingjob != null) {
+                        websocketpollingjob.cancel(true);
+                    }
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                            "Connection lost no ping from YIO DOCK");
+                    updateState(GROUP_OUTPUT, YIODOCKSTATUS, UnDefType.UNDEF);
+                }
+            }
+        };
+        websocketpollingjob = scheduler.scheduleWithFixedDelay(websocketpollingthread, 0, 30, TimeUnit.SECONDS);
+    }
+
+    protected boolean getbooleanauthenticationok() {
+        return yioremotedockhandler.booleanauthenticationok;
+    }
```

Comment:
Is there a reason this method is protected? You could also access the flag directly. I make a suggestion below.

Reply:
resolved